aea/helpers/base.py

```diff
@@ -201,21 +201,41 @@ def load_module(dotted_path: str, filepath: Path) -> types.ModuleType:
     return module
 
 
-def import_module(dotted_path: str, module_obj) -> None:
+def import_aea_module(dotted_path: str, module_obj) -> None:
     """
-    Add module to sys.modules.
+    Add an AEA module to sys.modules.
+
+    The parameter dotted_path has the form:
+
+        packages.<author_name>.<package_type>.<package_name>
+
+    If the closed-prefix packages are not present, add them to sys.modules.
+    This is done in order to emulate the behaviour of the true Python import system,
+    which in fact imports the packages recursively, for every prefix.
+
+    E.g. see https://docs.python.org/3/library/importlib.html#approximating-importlib-import-module
+    for an explanation on how the 'import' built-in function works.
 
     :param dotted_path: the dotted path to be used in the imports.
     :param module_obj: the module object. It is assumed it has been already executed.
     :return: None
     """
-    # if path is nested, and the root package is not present, add it to sys.modules
-    split = dotted_path.split(".")
-    if len(split) > 1 and split[0] not in sys.modules:
-        root = split[0]
-        sys.modules[root] = types.ModuleType(root)
 
-    # add the module at the specified path.
+    def add_namespace_to_sys_modules_if_not_present(dotted_path: str):
+        if dotted_path not in sys.modules:
+            sys.modules[dotted_path] = types.ModuleType(dotted_path)
+
+    # add all prefixes as 'namespaces', since they are not actual packages.
+    split = dotted_path.split(".")
+    assert len(split) > 3
```

Comment:
If this assert is not met I imagine the error message will not be very informative?

Reply:
Notice that the message of the `AssertionError` raised from this `assert` is only for the internals of the framework.  It is not related to the issue directly. But I agree with the comment, that assertion needs a message :-)